py/torch_tensorrt/dynamo/runtime/_PythonTorchTensorRTModule.py

```diff
@@ -182,7 +183,19 @@ def __init__(
         # Unused currently - to be used by Dynamic Shape support implementation
         self.memory_pool = None
 
-        self.serialized_engine = serialized_engine
+        if cuda_engine:
+            assert isinstance(
+                cuda_engine, trt.ICudaEngine
+            ), "Cuda engine must be a trt.ICudaEngine object"
+            self.engine = cuda_engine
+        elif serialized_engine:
+            assert isinstance(
+                serialized_engine, bytes
+            ), "Serialized engine must be a bytes object"
+            self.engine = serialized_engine
```

Comment:
Also do not overload the meaning of `self.engine`. Either it should be the serialized engine or the live engine not both